controller/integration/test/controlflow/controlflow_test.go

```diff
@@ -38,51 +39,55 @@ func Test_Finalizer_Integration_Controlflow(t *testing.T) {
 		}
 	}
 
-	resources := []controller.Resource{
-		controller.Resource(tr),
-	}
+	// We setup and start the controller.
+	var testWrapper wrapper.Interface
+	{
+		c := nodeconfig.Config{
+			Resources: []controller.Resource{
+				tr,
+			},
 
-	c := nodeconfig.Config{
-		Resources: resources,
+			Name:      operatorName,
+			Namespace: testNamespace,
+		}
 
-		Name:      operatorName,
-		Namespace: testNamespace,
-	}
+		testWrapper, err = nodeconfig.New(c)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
 
-	nodeconfigWrapper, err := nodeconfig.New(c)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
+		testWrapper.MustSetup(testNamespace)
+		defer testWrapper.MustTeardown(testNamespace)
+		newController := testWrapper.Controller()
+		go newController.Boot()
+		<-newController.Booted()
 	}
 
-	testWrapper := wrapper.Interface(nodeconfigWrapper)
-
-	testWrapper.MustSetup(testNamespace)
-	defer testWrapper.MustTeardown(testNamespace)
-
-	controller := testWrapper.Controller()
-
-	// We start the controller.
-	go controller.Boot()
-
 	// We create an object which is valid and wait for the framework to add a
 	// finalizer.
-	obj := &v1alpha1.NodeConfig{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      objName,
-			Namespace: testNamespace,
-		},
-	}
 	var createdObj interface{}
-	operation := func() error {
-		createdObj, err = testWrapper.CreateObject(testNamespace, obj)
+	{
+		o := func() error {
```

Comment:
The diff is weird but nothing changed. Just making the code more beautiful for my eyes so I could got into the matter.